Feature/propwidget

petersteneteg opened the pull request:
Disable and enable layout in the propertylists when doing many changes. 10X perf in some cases.
PropertyOwner got a clear() and empty() method.

petersteneteg committed:
OpenGL: use 32F for depth textures.

petersteneteg committed:
Python/Qtwidgets perf fixes

Disable and enable layout in the propertylists when doing many changes. !0X perf in some cases.
PropertyOwner got a clear() and empty() method.

petersteneteg committed:
OpenGL: Warning fix

ResearchDaniel reviewed (dismissed):
What’s the reason for changing the depth format to 32-bit?

petersteneteg commented:
We use a float32 on the cpu side, so to match that one. And I ran into some precision issues..

martinfalk commented on modules/python3/bindings/include/inviwopy/vectoridentifierwrapper.h:
```diff
@@ -59,11 +59,21 @@ class VectorIdentifierWrapper {
+            }
         } else {
-            throw pybind11::index_error();
+            const auto ind = static_cast<ptrdiff_t>(vector_.size()) + pos;
```
that's neat

martinfalk commented on modules/qtwidgets/src/propertylistwidget.cpp:
```diff
@@ -195,6 +205,16 @@ bool PropertyListWidget::event(QEvent* e) {
+void PropertyListWidget::onProcessorNetworkEvaluationBegin() { 
+    setUpdatesEnabled(false); 
+    listLayout_->setEnabled(false);
+}
```
this should take care of most of the issues we encountered so for, shouldn't it? Nice.

petersteneteg committed:
Python/Qtwidgets fixes

Python does not handle char enums well, unsinged char works.
Expose more python functionallity

petersteneteg committed:
Postprocessing: Regression update

Inviwo-Team committed:
Jenkins: Format fixes

petersteneteg commented:
@martinfalk I think this merge resulted in a quite substantial perf regression, My guess is that it is related to the move to 32 bit depth, does that make sense to you?

martinfalk commented:
Not sure. That would need performance testing in particular with different hardware. In some older posts I found that AMD GPUs did not support 24bits at all...

See also https://www.khronos.org/opengl/wiki/Common_Mistakes#Depth_Buffer_Precision

petersteneteg commented:
this for the ray caster with animation regression test:
![image](https://user-images.githubusercontent.com/3638222/129883409-50a05614-2baf-4f4c-ae2d-44d9114348ea.png)
the drop is when the bitdepth changed. see the last set of runs on http://jenkins.inviwo.org:8080/job/tests/job/hist/Regression/

But maybe we should consider other hardware too?
